Content.Server/Lathe/LatheSystem.cs

```diff
@@ -150,6 +150,14 @@ public static List<ProtoId<LatheRecipePrototype>> GetAllBaseRecipes(LatheCompone
             return component.StaticRecipes.Union(component.DynamicRecipes).ToList();
         }
 
+        public List<ProtoId<LatheRecipePrototype>> GetAllRecipes(EntityUid uid, LatheComponent component)
+        {
+            var baseRecipes = component.StaticRecipes.Union(component.DynamicRecipes);
+            if (!TryComp<EmagLatheRecipesComponent>(uid, out var emagRecipes))
+                return baseRecipes.ToList();
+            return baseRecipes.Union(emagRecipes.EmagStaticRecipes).Union(emagRecipes.EmagDynamicRecipes).ToList();
+        }
```

Comment:
okay i left some comments on your other PR and i didn't realize what the actual purpose of this was. This should still be accomplished with an event, or else it'll need updating for all recipes sources. But, you can totally just stick a boolean on the event for getting recipes and do it like that.